### PR TITLE
[BUG] Duplicate select form element #8147

### LINKED ISSUE #8147: [BUG] Duplicate select form element
### Description

These 2 components should be combined into 1

- `components/form/DropDown.js`
- `components/form/Select.js`

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mianakbarjan**
## Fixes Issue #8147 

Closes #8147 

## Changes proposed

[x] Deleted Dropdown.js form component
[x] Created a new Select form component including common and individual variables from both Dropdown and the previous Select component. 
[x] Replaced the Dropdown component in userEvents.js with the new Select component.
[x] Fixed the Manage Event and Manage Milestone Typos.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

No screenshots applicable as changes are only made to the code and there are no visible changes in the UI.

## Note to reviewers
Sorry for closing the previous pull request since the naming convention of the variables was not up to the mark. 

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/8220"></a>

### CODE REVIEW

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Storybook needs updating `stories/components/form/DropDown.stories.js`

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for your contribution but I seem to get many errors ...

EVENTS page

Screenshot 2023-07-18 at 07 56 32

ACCOUNT STATISTICS page


Screenshot 2023-07-18 at 07 57 02


Please test your changes before requesting a review

**Comment by mianakbarjan** [comment]
Hi @eddiejaoude, I've fixed the errors you got. However, I don't know further tests that I can do because the Select component is only visible on a few pages, where the tests are now successful. Sorry for any inconvenience caused since I'm new to open-source development and still learning. I will now request a review again.

File: components/form/Select.js [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
+        <label htmlFor="dropdown-select" className={className}>
           {label}
         </label>
       )}
       <select
-        id={name}
-        name={name}
-        className="mt-2 text-primary-high dark:bg-primary-high dark:text-white border-2 transition-all duration-250 ease-linear rounded px-6 py-2 mb-2 block w-full sm:text-sm sm:leading-6"
-        defaultValue={value}
+        id="dropdown-select"
```
Comment (eddiejaoude) [inline components/form/Select.js:17]: I think `id` should be the same as `name`

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for making the changes.

Some of the dropdowns don't work though...

Screenshot 2023-07-23 at 15 05 17

**Comment by mianakbarjan** [comment]
Hi @eddiejaoude, sorry for the poor PRs made previously. I have finally made all tests and fixed the dropdowns wherever they are visible in the frontend. They all seem to be working now. I will now request for changes again. Sorry again for the delay and the inconvenience caused.

**Comment by mianakbarjan** [comment]
The PR seems to have failed some tests, don't know exactly what they mean by the errors. If you want, I can try to look into them  but I'd appreciate it if I get some help.

**Comment by eddiejaoude** [comment]
> The PR seems to have failed some tests, don't know exactly what they mean by the errors. If you want, I can try to look into them but I'd appreciate it if I get some help.

The error message is `Ensures select element has an accessible name` with solutions...
- "message": "Form element does not have an implicit (wrapped) <label>",
- "message": "Form element does not have an explicit <label>",
- "message": "aria-label attribute does not exist or is empty",
- "message": "aria-labelledby attribute does not exist, references elements that do not exist or references elements that are - "message": "Element has no title attribute",
- "message": "Element's default semantics were not overridden with role=\"none\" or role=\"presentation\"",

> Fix any of the following:
    +   Form element does not have an implicit (wrapped) <label>
    +   Form element does not have an explicit <label>
    +   aria-label attribute does not exist or is empty
    +   aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
    +   Element has no title attribute

**Comment by mianakbarjan** [comment]
@eddiejaoude I've now added the label attribute to all the components, hopefully this time the tests will be successful. Requesting for a review again.

**Comment by eddiejaoude** [comment]
Thank you for making the changes 👍  There are still a couple of issues before we can merge the PR...

- please resolve the conflicts (the `navigation` file got renamed, please move your changes into the new file)
- `/events` page does not work

**Comment by mianakbarjan** [comment]
Hey @eddiejaoude, I don't know how to work this out since my local git repository and LinkFree's main repository have conflicts so I cannot change file names or make proper changes even though I've changed the local repository's filenames to try and fix it, but they won't get pushed. I don't know how to bring my forked repository and LinkFree's repository to the same level so that I can then work on it locally.

**Comment by mianakbarjan** [comment]
I have made the necessary changes and removed all conflicts, hopefully this time, the code will work fine.

**Comment by mianakbarjan** [comment]
It failed one test again, I've been trying to run the tests locally however I get the error tests/setup/global-setup.js saying fetch is not defined.

**Comment by dan-mba** [comment]
> It failed one test again, I've been trying to run the tests locally however I get the error tests/setup/global-setup.js saying fetch is not defined.

Make sure you are using Node 18 or newer.
Fetch was added in node 18.

**Comment by mianakbarjan** [comment]
Hopefully the errors are now fixed as all tests run successfully locally. A change had to be made in the events test file.

File: components/account/manage/Navigation.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (eddiejaoude) [inline components/account/manage/Navigation.js:1]: This change is not required?

File: components/user/UserEvents.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ export default function UserEvents({
     { value: "paid", name: "Paid Events" },
     { value: "past", name: "Past Events" },
   ];
-
+
```
Comment (eddiejaoude) [inline components/user/UserEvents.js:24]: This extra spaces is not required

File: components/user/UserEvents.js [hunk 4]
```diff
@@ -54,9 +54,8 @@ export default function UserEvents({
     }
     return events.filter((event) => filterByEventType(event, eventType));
   };
-
+
```
Comment (eddiejaoude) [inline components/user/UserEvents.js:57]: Same here

File: components/account/manage/Navigation.js (hunk 2 above)
Comment (eddiejaoude) [reply components/account/manage/Navigation.js:1]:
```suggestion
```

File: stories/components/form/DropDown.stories.js [hunk 5]
```diff
@@ -13,10 +13,10 @@ const options = [
...
   args: {
-    eventType: 3,
+    value: 3,
     label: 'Test',
     options: options,
-    handleEventTypeChange: (e) => console.log(e.target.value),
+    onChange: (e) => console.log(e.target.value),
     className: ''
   }
 };
```
Comment (eddiejaoude) [inline stories/components/form/DropDown.stories.js:22]: The `DropDown.stories.js` file should be renamed to `Select.stories.js` to match the component

File: components/user/UserEvents.js [hunk 6]
```diff
@@ -69,18 +68,25 @@ export default function UserEvents({
...
+          onChange={handleEventTypeChange}
+          options={filteredEventOptions.map((option) => ({
+            label: option.name,
+            value: option.value,
+          }))}
+          className="inline text-center text-sm font-medium leading-6 text-primary-high sm:pt-1.5"
         />
       )}
-
+
```
Comment (eddiejaoude) [inline components/user/UserEvents.js:89]:
```suggestion

```

File: components/user/UserEvents.js (hunk 3 above)
Comment (eddiejaoude) [reply components/user/UserEvents.js:24]:
```suggestion

```

File: components/user/UserEvents.js (hunk 4 above)
Comment (eddiejaoude) [reply components/user/UserEvents.js:56]:
```suggestion

```

File: components/Tabs.js [hunk 7]
```diff
@@ -16,21 +16,16 @@ export default function Tabs({ tabs, setTabs }) {
   return (
     <div>
       <div className="sm:hidden">
-        <label htmlFor="tabs" className="sr-only">
-          Select a tab
-        </label>
+
```
Comment (eddiejaoude) [inline components/Tabs.js:19]:
```suggestion
```

File: components/Tabs.js [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import Link from "@components/Link";
+import Select from "@components/form/Select"
```
Comment (eddiejaoude) [inline components/Tabs.js:2]: Please run the linter, missing semi colon to match our project coding standards

```suggestion
import Select from "@components/form/Select";
```

File: components/form/Select.js [hunk 9]
```diff
@@ -1,26 +1,35 @@
...
       <select
-        id={name}
-        name={name}
-        className="mt-2 text-primary-high dark:bg-primary-high dark:text-white border-2 transition-all duration-250 ease-linear rounded px-6 py-2 mb-2 block w-full sm:text-sm sm:leading-6"
-        defaultValue={value}
+        id={value}
+        value={value}
+        onChange={onChange}
+        className={`mt-2 text-primary-high dark:bg-primary-high dark:text-white border-2 transition-all duration-250 ease-linear rounded px-6 py-2 mb-2 block w-full sm:text-sm sm:leading-6 ${
+          restProps.className || ''
```
Comment (eddiejaoude) [inline components/form/Select.js:21]: Does not match our project coding standards, double quotes required

```suggestion
          restProps.className || ""
```

File: components/user/UserEvents.js [hunk 10]
```diff
@@ -69,18 +66,24 @@ export default function UserEvents({
...
+      )}
+  
       {!manage && (
-        <DropdownMenu
-          eventType={eventType}
-          handleEventTypeChange={handleEventTypeChange}
-          options={filteredEventOptions}
-          label="Select Event Type:"
-          className="inline text-center text-sm font-medium leading-6 text-primary-high sm:pt-1.5 dark:text-primary-low"
+
```
Comment (eddiejaoude) [inline components/user/UserEvents.js:74]:
```suggestion
```

**Review by eddiejaoude — APPROVED** [review]
Looks good, thank you 👍

**Comment by mianakbarjan** [comment]
Thank you for giving me the opportunity Eddie! This was my first ever pull request that got merged and hopefully, I'll keep contributing to LinkFree!!!

**Pull request merged**